### PR TITLE
Contributing guide

### PR DESCRIPTION
**Pull request by BenFradet**

### CODE REVIEW

File: docs/src/main/tut/contributing.md [hunk 1]
```diff
@@ -0,0 +1,386 @@
...
+      // ...
+      case ListStatuses(owner, repo, ref, accessToken) =>
+        repos.listStatus(accessToken, headers, owner, repo, ref)
+      // ...
+    }
+  }
+}
+```
+
+## Wrapper
```
Comment (BenFradet) [inline docs/src/main/tut/contributing.md]: I don't what happened but the `Wrapper` section is offset almost centered.

File: docs/src/main/tut/contributing.md [hunk 2]
```diff
@@ -0,0 +1,386 @@
+---
+layout: docs
```
Comment (juanpedromoreno) [inline docs/src/main/tut/contributing.md]: I'm wondering if it makes sense to put this awesome contributing guide in the microsite main page, using `page` layout. We did it in cats in this way: http://typelevel.org/cats/contributing.html

What do you think?

File: docs/src/main/tut/contributing.md [hunk 3]
```diff
@@ -0,0 +1,386 @@
...
+
+## Token
+
+The first step needed to run the tests is a valid token which we can provide through an
+environment variable:
+
+```bash
+export GITHUB4S_ACCESS_TOKEN=aaaa
+```
+
```
Comment (juanpedromoreno) [inline docs/src/main/tut/contributing.md]: I'd add a link indicating how generate this personal access token

File: docs/src/main/tut/contributing.md [hunk 4]
```diff
@@ -0,0 +1,386 @@
...
+## Integration tests
+
+The integration tests are grouped by API in [github4s.integration package][integ-pkg]. As a result,
+we'll be writing our tests in [GHReposSpec][repos-integ-spec-scala]:
```
Comment (juanpedromoreno) [inline docs/src/main/tut/contributing.md]: Let's add that we only want integration tests for read-only operations

Comment (BenFradet) [inline docs/src/main/tut/contributing.md]: the funny thing is that I was planning on adding it but completely forgot about it :smile:

File: docs/src/main/tut/contributing.md (hunk 2 above)
Comment (BenFradet) [inline docs/src/main/tut/contributing.md]: Good idea, that way we avoid cluttering the documentation too much.

Comment (BenFradet) [inline docs/src/main/tut/contributing.md]: Should we move it to a `CONTRIBUTING.md` at the root of the repo as in cats?

Comment (juanpedromoreno) [inline docs/src/main/tut/contributing.md]: We could, however, it's not mandatory. If we want to move it to the root I'm fine with that. Nevertheless, I'd separate it from the current `CONTRIBUTING.md` file in the root. What do you think about `DEV_GUIDE.md` or something similar? (afterward, when it's copied to the microsite territory we can rename it on the fly).

Comment (BenFradet) [inline docs/src/main/tut/contributing.md]: I'm fine with leaving it where it is for now.

**Pull request merged**